datafusion/functions-aggregate/src/stddev.rs

```diff
@@ -0,0 +1,380 @@
+// specific language governing permissions and limitations
+// under the License.
+
+//! Defines physical expressions that can evaluated at runtime during query execution
+
+use std::any::Any;
+use std::fmt::{Debug, Formatter};
+
+use arrow::{array::ArrayRef, datatypes::DataType, datatypes::Field};
+
+use datafusion_common::{internal_err, not_impl_err, Result};
+use datafusion_common::{plan_err, ScalarValue};
+use datafusion_expr::function::{AccumulatorArgs, StateFieldsArgs};
+use datafusion_expr::utils::format_state_name;
+use datafusion_expr::{Accumulator, AggregateUDFImpl, Signature, Volatility};
+use datafusion_physical_expr_common::aggregate::stats::StatsType;
+
+use crate::variance::VarianceAccumulator;
+
+make_udaf_expr_and_func!(
+    Stddev,
+    stddev,
+    expression,
+    "Compute the standard deviation of a set of numbers",
+    stddev_udaf
+);
+
+/// STDDEV and STDDEV_SAMP (standard deviation) aggregate expression
+pub struct Stddev {
+    signature: Signature,
+    alias: Vec<String>,
+}
+
+impl Debug for Stddev {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Stddev")
+            .field("name", &self.name())
+            .field("signature", &self.signature)
+            .finish()
+    }
+}
+
+impl Default for Stddev {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Stddev {
+    /// Create a new STDDEV aggregate function
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::numeric(1, Volatility::Immutable),
+            alias: vec!["stddev_samp".to_string()],
+        }
+    }
+}
+
+impl AggregateUDFImpl for Stddev {
+    /// Return a reference to Any that can be used for downcasting
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &str {
+        "stddev"
+    }
+
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
+        if !arg_types[0].is_numeric() {
+            return plan_err!("Stddev requires numeric input types");
+        }
+
+        Ok(DataType::Float64)
+    }
+
+    fn state_fields(&self, args: StateFieldsArgs) -> Result<Vec<Field>> {
+        Ok(vec![
+            Field::new(
+                format_state_name(args.name, "count"),
+                DataType::UInt64,
+                true,
+            ),
+            Field::new(
+                format_state_name(args.name, "mean"),
+                DataType::Float64,
+                true,
+            ),
+            Field::new(format_state_name(args.name, "m2"), DataType::Float64, true),
+        ])
+    }
+
+    fn accumulator(&self, acc_args: AccumulatorArgs) -> Result<Box<dyn Accumulator>> {
+        if acc_args.is_distinct {
+            return not_impl_err!("STDDEV_POP(DISTINCT) aggregations are not available");
+        }
+        Ok(Box::new(StddevAccumulator::try_new(StatsType::Sample)?))
+    }
+
+    fn create_sliding_accumulator(
```

Comment:
I think we could rely on the default impl, since they are equivalent

Reply:
https://github.com/apache/datafusion/blob/6b7021479c956ba3ca2a04fff81487cd57f80624/datafusion/expr/src/udaf.rs#L414-L419

Reply:
Thanks @jayzhan211,  It's better.